qcodes/instrument/parameter.py

```diff
@@ -1308,29 +1331,45 @@ def raw_value(self) -> ParamRawDataType:
             This bug will not be fixed since the `raw_value` property will be
             removed soon.
             """
-            return self._parameter.source.cache._value
-
-        @property
-        def _value(self) -> ParamDataType:
-            return self._parameter._from_raw_value_to_value(self.raw_value)
+            if self._parameter.source is None:
+                raise TypeError("Cannot get the raw value of a "
+                                "DelegateParameter that delegates to None")
+            return self._parameter.source.cache.get(get_if_invalid=False)
 
         @property
         def max_val_age(self) -> Optional[float]:
+            if self._parameter.source is None:
+                return None
             return self._parameter.source.cache.max_val_age
 
         @property
         def timestamp(self) -> Optional[datetime]:
+            if self._parameter.source is None:
+                return None
             return self._parameter.source.cache.timestamp
 
         def get(self, get_if_invalid: bool = True) -> ParamDataType:
+            if self._parameter.source is None:
+                raise TypeError("Cannot get the cache of a "
+                                "DelegateParameter that delegates to None")
             return self._parameter._from_raw_value_to_value(
                 self._parameter.source.cache.get(get_if_invalid=get_if_invalid))
 
         def set(self, value: ParamDataType) -> None:
+            if self._parameter.source is None:
+                raise TypeError("Cannot set the cache of a DelegateParameter "
+                                "that delegates to None")
             self._parameter.validate(value)
             self._parameter.source.cache.set(
                 self._parameter._from_value_to_raw_value(value))
 
+        def _set_from_raw_value(self, value: ParamRawDataType) -> None:
+            if self._parameter.source is None:
+                raise TypeError("Cannot set the cache of a DelegateParameter "
+                                "that delegates to None")
+            self._parameter.validate(value)
```

Comment:
as per my memory, _set_from_raw_value was made with an assumtion that it will be used in a "get" context (as opposed to "set"). this is why in _BaseParameter, _set_from_raw_value 
 validates the value only if _validate_on_get is True. I don't think that this assumtion is a good one. So either the _validate_on_get  condition needs to be gone but then we need to be careful about validation values in group parameter case, or add the _validate_on_get  condition below for consistency just "for now":
```python
if self._parameter._validate_on_get:
    self._parameter.validate(value)
```

what do you think?

Reply:
Actually this never matters. It calls set on the base parameter which will always do a validate. In anycase it should validate against the value and not the raw_value. So lets perhaps just remove this validator